data-processing/tests/test_extract_definitions.py

```diff
@@ -123,32 +95,84 @@ def test_extract_symbol_nickname_pairs(text, tex, gold):
         ), "Nickname Incorrect"
 
 
-@pytest.mark.parametrize(
-    "text,gold",
-    [
-        # Case 1: Abbreviation in parentheses
-        (
-            "We use a Convolutional Neural Network (CNN) based architecture in this model, which is an improvement over state-of-the-art.",
-            [
-                (
-                    "CNN",
-                    "Convolutional Neural Network",
-                )
-            ],
-        ),
-        # Case 2: Abbreviation contains multiple lowercase letters
-        (
-            "We propose a new class of architectures called Conductive Networks (CondNets) in this paper.",
-            [
-                (
-                    "CondNets",
-                    "Conductive Networks",
-                )
-            ],
-        ),
-    ],
-)
-def test_extract_abbreviation_expansion_pairs(text, gold):
+def test_extract_symbol_nickname_pairs_case_2():
+    # Case 2: Nickname after symbol.
+    text = "The architecture consists of SYMBOL dense layers trained with SYMBOL learning rate."
+    tex = "The architecture consists of [[FORMULA:L_d]] dense layers trained with [[FORMULA:\alpha]] learning rate."
+    gold = [("L_d", "dense layers"), ("\alpha", "learning rate")]
+    featurized_text = model.featurize(text)
+    features = [featurized_text]
+    symbol_texs = get_symbol_texs(text, tex)
+    symbol_nickname_pairs = get_symbol_nickname_pairs(text, features[0], symbol_texs)
+    assert len(symbol_nickname_pairs) == len(
+        gold
+    ), "Incorrect number of Symbol-Nicknames"
+    for prediction_pair, gold_pair in zip(symbol_nickname_pairs, gold):
+        if symbol_texs is not None and prediction_pair.term_start in symbol_texs:
+            assert (
+                symbol_texs[prediction_pair.term_start] == gold_pair[0]
+            ), "Symbol Incorrect"
+        else:
+            assert (
+                text[prediction_pair.term_start : prediction_pair.term_end]
+                == gold_pair[0]
+            ), "Symbol Incorrect"
+        assert (
+            text[prediction_pair.definition_start : prediction_pair.definition_end]
+            == gold_pair[1]
+        ), "Nickname Incorrect"
```

Comment:
This part might make sense to put in a function of its own so it can be easily reused, but it's not critical.